src/iologindata.cpp

```diff
@@ -832,6 +841,31 @@ bool IOLoginData::savePlayer(Player* player)
 		return false;
 	}
 
+	// save account storage values
+	if (!db.executeQuery(fmt::format("DELETE FROM `account_storage` WHERE `account_id` = {:d}", player->getAccount()))) {
+		return false;
+	}
+
+	const auto& accountStorageMap = g_game.accountStorageMap.find(player->getAccount());
+	if (accountStorageMap == g_game.accountStorageMap.end()) {
+		return false;
+	}
+
+	DBInsert accountStorageQuery("INSERT INTO `account_storage` (`account_id`, `key`, `value`) VALUES");
+	for (const auto& it : accountStorageMap->second) {
+		if (!accountStorageQuery.addRow(fmt::format("{:d}, {:d}, {:d}", player->getAccount(), it.first, it.second))) {
+			return false;
+		}
+	}
+
+	if (!accountStorageQuery.execute()) {
+		return false;
+	}
+
+	if (g_game.getNumberOfPlayersByAccount(player->getAccount()) == 1) {
+		g_game.accountStorageMap.erase(player->getAccount());
```

Comment:
This is not a place to do it, as I have mentioned in my other comment. Calling this method does not mean that player is logging out. You are erasing the storage with every save of that player (player:save(), saveServer()).

Reply:
True. I missed that one (twice, lmao).